storage/src/mmr/verification.rs

```diff
@@ -90,6 +90,58 @@ impl Proof {
         }
         *root_hash == mmr_hasher.root_hash(self.size, peak_hashes.iter())
     }
+
+    /// Return the maximum size in bytes of any serialized `Proof`.
+    pub fn max_serialization_size<H: CHasher>() -> usize {
+        9 + (u8::MAX as usize * H::len())
+    }
+
+    /// Canonically serializes the `Proof` as:
+    /// ```text
+    ///    [0-8): size (u64 big-endian)
+    ///    [9-10): number of hashes (u8)
+    ///    [11-...): raw bytes of each hash, each of length `H::len()`
+    /// ```
+    pub fn serialize<H: CHasher>(&self) -> Vec<u8> {
+        let bytes_len = 8 + 1 + (self.hashes.len() * H::len());
```

Comment:
I've been using `size_of::<u64>()` rather than floating values for sizes.

Reply:
changed